src/cli/java/org/commcare/util/screen/SyncScreen.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    @Override
+    public void init (SessionWrapper sessionWrapper) throws CommCareSessionException {
+        this.sessionWrapper = sessionWrapper;
+        String command = sessionWrapper.getCommand();
+        Entry commandEntry = sessionWrapper.getPlatform().getEntry(command);
+        if (commandEntry instanceof RemoteRequestEntry) {
+            PostRequest syncPost = ((RemoteRequestEntry)commandEntry).getPostRequest();
+            Response response = makeSyncRequest(syncPost);
+            if (!response.isSuccessful()) {
+                throw new CommCareSessionException(
+                        String.format("Sync request failed with code %s and message %s.",
+                                response.code(),
+                                response.message()
+                        )
+                );
+            }
+            printStream.println(String.format("Sync successful with response %s", response));
+            printStream.println("Press 'enter' to continue.");
+        } else {
+            // expected a sync entry; clear session and show vague 'session error' message to user
+            throw new CommCareSessionException("Initialized sync request while not on sync screen");
+        }
+    }
+
+    private static String buildUrl(String baseUrl) {
+        HttpUrl.Builder urlBuilder = HttpUrl.parse(baseUrl).newBuilder();
+        return urlBuilder.build().toString();
+    }
+
+    private static MultipartBody buildPostBody(Hashtable<String, String> params) {
+        MultipartBody.Builder requestBodyBuilder = new MultipartBody.Builder()
+                .setType(MultipartBody.FORM);
+        // Add buffer param since this is necessary for some reason
+        requestBodyBuilder.addFormDataPart("buffer", "buffer");
+        for (String key: params.keySet()) {
+            requestBodyBuilder.addFormDataPart(key, params.get(key));
+        }
+        return requestBodyBuilder.build();
+    }
+
+    private Response makeSyncRequest(PostRequest syncPost) throws CommCareSessionException {
+        Hashtable<String, String> params = syncPost.getEvaluatedParams(sessionWrapper.getEvaluationContext());
+        String url = buildUrl(syncPost.getUrl().toString());
+        printStream.println(String.format("Syncing with url %s and parameters %s", url, params));
+        MultipartBody postBody = buildPostBody(params);
+        String credential = Credentials.basic(username, password);
+
+        Request request = new Request.Builder()
+                .url(url)
+                .method("POST", RequestBody.create(null, new byte[0]))
+                .header("Authorization", credential)
+                .post(postBody)
+                .build();
+        try {
+            OkHttpClient client = new OkHttpClient();
+            return client.newCall(request).execute();
+        } catch (IOException e) {
+            throw new CommCareSessionException("Exception while making sync request", e);
```

Comment:
So in theory this IOException could easily represent a number of different outcomes, right? A session exception's going to kill the session entirely. I'd imagine that it would make more sense for this to be a retry-able event which clearly expresses the error (in case it's an error that would appear on the phone and they are debugging)

Reply:
Good call, thanks!